### ISSUE TITLE
[Bug]: There is a significant difference in filter performance between Fabric.js version 6 and version 5.

### ISSUE THREAD

**Issue by luo-sb** [fabricjs/fabric.js#10099]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.2

### In What environments are you experiencing the problem?

Chrome

### Node Version (if applicable)

None

### Link To Reproduction

None

### Steps To Reproduce

I tried to use `fabric.filters.RemoveColor` in Fabric.js to remove the green screen from a video and then play it on the canvas. In version 5, the performance was as expected, and the playback was smooth. However, in version 6, it became very laggy. I discovered that the `fabric.Image.applyFilters()` method takes only 10ms to execute in version 5, but it takes 350ms-450ms in version 6.

Is this a bug, or is there something wrong with my code?

"This zip file contains the two versions of test demos that I wrote."
[fabricjs-demo.zip](https://github.com/user-attachments/files/16778823/fabricjs-demo.zip)

### Expected Behavior

It becomes as smooth as it was in version 5.

### Actual Behavior

.

**Comment by asturur** [comment]
Hopefully is a bug we can fix.
Is the difference visible in a single frame? or just on continued use? do you have any idea?

**Comment by luo-sb** [comment]
> Hopefully is a bug we can fix. Is the difference visible in a single frame? or just on continued use? do you have any idea?

Yes, even executing fabric.Image.applyFilters() only once takes about 350ms.

**Comment by asturur** [comment]
This was fixed, it was a bug indeed.
The fix is in 6.6.1
